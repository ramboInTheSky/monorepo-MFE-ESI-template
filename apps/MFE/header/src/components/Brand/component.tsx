import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"

export const Container = styled.div`
    margin: ${props => props.theme.styles.Brand.xs.margin};

    a,
    img {
        display: block;
        width: 100%;
        height: ${props => props.theme.styles.Brand.xs.height};
        width: ${props => props.theme.styles.Brand.xs.width};
        font-size: 0.9375rem;
        @media (min-width: ${breakpoints.values.md}px) {
            height: ${props => props.theme.styles.Brand.md.height};
            width: ${props => props.theme.styles.Brand.md.width};
        }

        @media (min-width: ${breakpoints.values.lg}px) {
            height: ${props => props.theme.styles.Brand.lg.height};
            width: ${props => props.theme.styles.Brand.lg.width};
            margin: ${props => props.theme.styles.Brand.lg.margin};
        }
    }
`
