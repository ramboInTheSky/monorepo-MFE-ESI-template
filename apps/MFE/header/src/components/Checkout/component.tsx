import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"

export type ContainerProps = {
    isinternationalcountry: boolean
}

export const Container = styled.div<ContainerProps>`
    @media (min-width: ${breakpoints.values.md}px) {
        width: ${props => props.theme.styles.Checkout.md.width};
        height: ${props => props.theme.styles.Checkout.md.height};
    }

    @media (min-width: ${breakpoints.values.lg}px) {
        width: ${props => props.theme.styles.Checkout.lg.width};
        height: ${props => props.theme.styles.Checkout.lg.height};
    }

    @media (min-width: ${breakpoints.values.xl}px) {
        width: ${props => (props.isinternationalcountry ? "10rem" : props.theme.styles.Checkout.xl.width)};
        height: ${props => props.theme.styles.Checkout.xl.height};
    }
`
