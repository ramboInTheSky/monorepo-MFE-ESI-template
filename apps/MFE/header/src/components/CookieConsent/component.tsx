import {Typography} from "@mui/material"
import styled from "styled-components"

export const Container = styled.div`
    position: fixed;
    bottom: 0.625rem;
    right: 0.625rem;
    width: 16.5rem;
    padding: 1rem;
    background-color: ${({theme}) => theme.colours.form.default.background};
    border: ${({theme}) => theme.colours.utilities.divider};
    border-radius: 0.25rem;
`

export const StyledTitle = styled(Typography)`
    padding-bottom: 0.35rem;
`

export const Link = styled.a`
    color: ${({theme}) => theme.colours.text.default};
    text-decoration: underline;
`

export const StyledIcon = styled.img`
    position: absolute;
    right: 0.625rem;
    top: 0.625rem;
    cursor: pointer;
`
