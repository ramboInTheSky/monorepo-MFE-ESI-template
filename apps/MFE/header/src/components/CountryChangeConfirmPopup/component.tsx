import {Typography, Button} from "@mui/material"
import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"

export const Container = styled.div`
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);

    width: 18rem;
    height: 18.1875rem;
    padding: 1rem;
    background-color: ${({theme}) => theme.colours.form.default.background};
    border: ${({theme}) => theme.colours.utilities.divider};
    border-radius: 0.25rem;

    @media (min-width: ${breakpoints.values.md}px) {
        width: 35rem;
        height: 13.1875rem;
    }
`

export const StyledTitle = styled(Typography)`
    width: 16rem;
    height: 1.625rem;
    margin: 0 0 0.4375rem;
    font-size: 1.0625rem;
    font-weight: bold;
    line-height: 1.5;
    letter-spacing: normal;
`

export const StyledText = styled(Typography)`
    width: 16rem;
    margin: 0.4375rem 0 1rem;
    font-size: 0.875rem;
    font-weight: normal;
    font-style: normal;
    line-height: 1.43;
    letter-spacing: 0.0075rem;

    @media (min-width: ${breakpoints.values.md}px) {
        width: 34rem;
    }
`

export const StyledButtonWrapper = styled.div`
    display: block;

    @media (min-width: ${breakpoints.values.md}px) {
        display: flex;
        flex-direction: row-reverse;
    }
`

export const StyledButtonConfirm = styled(Button)`
    width: 100%;
    font-size: 0.75rem;
    font-weight: ${({theme}) => theme.colours.font.primary.medium.weight};
    letter-spacing: 0.0625rem;
    margin: 0rem 0rem 0.75rem;
    border: 0px;
    color: ${({theme}) => theme.colours.form.buttonPrimary.color};
    opacity: 1;
    border-radius: 0.25rem;
    background: ${({theme}) => theme.colours.form.buttonPrimary.background};
    padding: 0.5625rem 0.5rem;

    @media (min-width: ${breakpoints.values.md}px) {
        width: 12.5rem;
        height: 2.75rem;
        margin: 0rem 1rem;
    }

    &:hover,
    &:active,
    &:focus {
        background: ${({theme}) => theme.colours.form.buttonPrimary.background};
        border: ${({theme}) => theme.colours.form.buttonPrimary.border};
        color: ${({theme}) => theme.colours.form.buttonPrimary.color};
    }
`

export const StyledButtonCancel = styled(Button)`
    width: 100%;
    font-size: 0.75rem;
    font-weight: ${({theme}) => theme.colours.font.primary.medium.weight};
    letter-spacing: 0.0625rem;
    margin: 0px;
    border: 0px;
    color: ${({theme}) => theme.colours.form.buttonSecondary.color};
    opacity: 1;
    border-radius: 0.25rem;
    border: 1px solid black;
    background: ${({theme}) => theme.colours.form.buttonSecondary.background};
    padding: 0.5625rem 0.5rem;

    @media (min-width: ${breakpoints.values.md}px) {
        width: 12.5rem;
        height: 2.75rem;
    }
`
