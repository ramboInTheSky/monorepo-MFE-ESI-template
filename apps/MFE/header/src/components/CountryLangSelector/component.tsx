import styled from "styled-components"
import {CTA} from "@monorepo/cta"
import {breakpoints} from "@monorepo/themes"
import {Typography, Button as MUIButton, Drawer as MUIDrawer} from "@mui/material"
import {COUNTRY_SELECTOR_DRAWER_BREAKPOINT} from "../../config/constants"

export type CountryFlagImgProps = {
    backgroundImg: string
}

export type StyledIcon = {
    textAlignment?: string
}

export const DrawerContainer = styled(MUIDrawer)`
    position: initial;
    && > div:first-child {
        background: rgba(0, 0, 0, 0.9);
    }

    && > div:nth-child(3) {
        // to fix Safari v13 - as the dropdown menu is hidden with the drawer component overflow-y set as auto
        overflow-y: initial;
        width: 100%;
        @media (min-width: ${COUNTRY_SELECTOR_DRAWER_BREAKPOINT}) {
            overflow-y: auto;
            width: ${props => props.theme.styles.DrawerContainer.xs.width};
        }
    }
`

export const Wrapper = styled.div`
    height: 2.75rem;
    font-size: 0.9375rem;
`

export const StyledIcon = styled.img<StyledIcon>`
    transform: ${({textAlignment}) => `rotate(${textAlignment === "rtl" ? "90" : "270"}deg)`};
    margin: -0.125rem 0 0 0.25rem;
    top: 0;
`

export const ROWLink = styled.a`
    font-family: ${props => `${props.theme.colours.font.primary.regular.family}, ${props.theme.colours.font.default}`};
    color: ${props => props.theme.colours.text.hyperlink};
    text-decoration: underline;
    font-size: 0.875rem;
    &:visited {
        color: ${props => props.theme.colours.text.hyperlink};
    }
`

export const CloseButton = styled(MUIButton)`
    && {
        width: 4.5rem;
        background: ${({theme}) => theme.colours.utilities.backgroundAccent};
        color: ${({theme}) => theme.colours.text.hyperlink};
        font-family: ${({theme}) => `${theme.colours.font.primary.medium.family}, ${theme.colours.font.default}`};
        border-radius: 0;
        border: 0;

        &:hover,
        &:active {
            background: ${({theme}) => theme.colours.utilities.backgroundAccent};
            border: 0;
        }

        &:focus {
            border: 0;
            border-radius: 0;
        }
        span {
            font-family: ${({theme}) => `${theme.colours.font.primary.medium.family}, ${theme.colours.font.default}`};
        }
    }
`

export const Container = styled.button`
    cursor: pointer;
    padding: ${props => props.theme.styles.CountrySelector.Container.xs.padding};
    background-color: transparent;
    border: none;
    &:active {
        background-color: transparent;
        border: none;
    }
`

export const CountrySelectorROW = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 0.5rem 0;
    line-height: 1.25rem;
`

export const CountryFlagImg = styled.img`
    display: flex;
    border: ${({theme}) => {
        return `${
            theme.styles.CountrySelector.flagImg?.borderWidth
                ? theme.styles.CountrySelector.flagImg.borderWidth
                : "0.125rem"
        } solid ${theme.colours.header.navUpperBackground.color}`
    }};
    border-radius: 50%;

    width: 1.375rem;
    height: 1.375rem;
    background: ${({theme}) => theme.colours.header.navUpperBackground.color};
    @media (min-width: ${breakpoints.values.md}px) {
        width: ${({theme}) => theme.styles.CountrySelector.flagImg?.width || "1.5rem"};
        height: ${({theme}) => theme.styles.CountrySelector.flagImg?.height || "1.5rem"};
    }
`

export const CountrySelectorContainer = styled.div`
    display: flex;
    flex-direction: column;
    background: ${({theme}) => theme.colours.text.reversed};
    height: 21.75rem;
    width: 100%;
    @media (min-width: ${COUNTRY_SELECTOR_DRAWER_BREAKPOINT}) {
        width: ${props => props.theme.styles.DrawerContainer.xs.width};
        height: 100%;
    }
`

export const CountrySelectorContainerHeader = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding-left: 1rem;
    height: 2.75rem;
    background: ${({theme}) => theme.colours.utilities.backgroundAccent};
`

export const CountrySelectorContainerElement = styled.div`
    display: flex;
    flex-direction: column;
    padding: 0.5rem 1rem 0;
    @media (min-width: ${COUNTRY_SELECTOR_DRAWER_BREAKPOINT}) {
        padding: 0.5rem 1rem;
    }
`
export const CountrySelectorCTAWrapper = styled.div`
    display: flex;
    margin: 0.5rem 1rem;
    height: 2.75rem;
    @media (min-width: ${breakpoints.values.lg}px) {
        font-size: 0.875rem !important;
        height: 2.25rem;
    }
`

export type CountrySelectorContainerLanguageButtonsProps = {
    selected?: boolean
}

export const CountrySelectorContainerLanguageButtons = styled.button<CountrySelectorContainerLanguageButtonsProps>`
    display: flex;
    justify-content: center;
    align-items: center;
    border: ${({theme, selected}) =>
        selected ? theme.colours.form.selectedAccent.border : theme.colours.form.default.border};
    background: ${({theme, selected}) =>
        selected ? "rgba(48, 167, 75, 0.05)" : theme.colours.form.buttonSecondary.background};
    font-weight: ${({theme, selected}) =>
        selected ? theme.colours.font.primary.medium.weight : theme.colours.font.primary.light.weight};
    border-radius: 0.25rem;
    font-family: ${({theme, selected}) =>
        selected
            ? `${theme.colours.font.primary.medium.family}, ${theme.colours.font.default}`
            : `${theme.colours.font.primary.regular.family}, ${theme.colours.font.default}`};
    width: calc((100% - 1rem) / 2);
    height: 2.75rem;
    font-size: 0.875rem;
    @media (min-width: ${breakpoints.values.lg}px) {
        height: 2rem;
    }
    cursor: pointer;
    &:focus {
        outline: none;
    }
`

export const Title = styled(Typography)`
    letter-spacing: 1px;
`

export const CountrySelectorContainerLanguage = styled.div`
    display: flex;
    justify-content: space-between;
    padding: 0.5rem 0;
`

export const StayText = styled(Typography).attrs({component: "span"})`
    && {
        color: ${({theme}) => theme.colours.text.hyperlink};
        font-family: ${({theme}) => `${theme.colours.font.primary.regular.family}, ${theme.colours.font.default}`};
        font-weight: ${({theme}) => theme.colours.font.primary.medium.weight};
        padding-right: 0.25rem;
    }
`
export const CountrySelectorBottomText = styled.div`
    display: flex;
    justify-content: flex-end;
    padding: 0.75rem 1rem 0;
`
export const StayButton = styled.button`
    cursor: pointer;
    background-color: transparent;
    border: none;
    &:active {
        background-color: transparent;
        border: none;
    }
`
export const CTAButton = styled(CTA)`
    font-size: 0.875rem !important;
`
