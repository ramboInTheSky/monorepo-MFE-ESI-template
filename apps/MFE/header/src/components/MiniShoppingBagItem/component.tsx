import styled from "styled-components"

export const ItemWrapper = styled.div`
    display: flex;
    border-top: ${props => props.theme.colours.utilities.divider};
`
