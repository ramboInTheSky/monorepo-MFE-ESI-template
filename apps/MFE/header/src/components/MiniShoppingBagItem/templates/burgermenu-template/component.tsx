import styled from "styled-components"
import {IN_STOCK, SOLD_OUT} from "../../../../config/constants"

export const ProductDetails = styled.div`
    display: grid;
    grid-template-columns: repeat(2, minmax(0, 1fr));
    width: 100%;
    margin-right: 16px;
    font-size: 12px;

    .mt-5px {
        margin-top: 5px;
    }
    .block {
        display: block;
    }
`

export const ProductNameH = styled.h2`
    font-weight: bold;
    font-size: 12px;
    margin-top: 1rem;
`
export const SearchDescription = styled.div`
    color: ${({theme: {colours}}) => colours.text.muted};
`
export const Price = styled.p`
    font-weight: bold;
    text-align: right;
    margin-top: 1rem;
`

export const StatusWrapper = styled.span<{status: string}>`
    font-weight: bold;
    font-size: 12px;
    text-align: right;
    display: block;
    margin-top: 18px;
    color: ${props => {
        switch (props.status) {
            case IN_STOCK:
                return "black"
            case SOLD_OUT:
                return props.theme.colours.text.error
            default:
                return "black"
        }
    }};
`
