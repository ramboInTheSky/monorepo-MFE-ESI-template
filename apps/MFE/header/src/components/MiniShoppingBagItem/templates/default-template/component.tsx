import styled from "styled-components"
import {IN_STOCK, SOLD_OUT} from "../../../../config/constants"

export const PriceParentWrapper = styled.div`
    white-space: nowrap;
    padding-right: 0.938rem;
`

export const StatusWrapper = styled.div<{status: string}>`
    font-family: ${props => `${props.theme.colours.font.primary.regular.family}, ${props.theme.colours.font.default}`};
    margin: 0.25rem 0 0 0;
    font-size: 0.875rem;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    color: ${props => {
        switch (props.status) {
            case IN_STOCK:
                return props.theme.colours.text.success
            case SOLD_OUT:
                return props.theme.colours.text.error
            default:
                return props.theme.colours.text.warning
        }
    }};
`

export const Item = styled.p`
    font-size: 12px;
    font-weight: bold;
    text-align: right;
`
export const ProductDetails = styled.div`
    padding: 1rem 0;
    width: 100%;

    && > p {
        font-family: ${props =>
            `${props.theme.colours.font.primary.regular.family}, ${props.theme.colours.font.default}`};
        color: ${props => props.theme.colours.text.muted};
        word-break: break-all;
        padding-right: 0.613rem;
    }
`
export const ProductDetailItem = styled.div`
    display: flex;
    justify-content: space-between;
    > h3:first-child {
        word-wrap: break-word;
        max-width: 10.3125rem;
        padding-right: 0.625rem;
    }
    > h3:last-child {
        white-space: nowrap;
    }
`
