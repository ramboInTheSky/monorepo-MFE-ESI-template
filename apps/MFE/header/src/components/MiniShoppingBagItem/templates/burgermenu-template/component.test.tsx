import React from "react"
import {render} from "@testing-library/react"
import {SCThemeProvider} from "@monorepo/theme-provider"
import {ProductDetails, StatusWrapper} from "./component"
import {mockTheme} from "../../../../../__mocks__/mockStore"
import {IN_STOCK} from "../../../../config/constants"

describe(" components: GridW100PR16 ", () => {
    it("should match the snapshot ", () => {
        const {asFragment} = render(<ProductDetails />)
        expect(asFragment()).toMatchSnapshot()
    })
})

describe("StatusWrapper ", () => {
    it("should match the snapshot with In stock status", () => {
        const {asFragment} = render(
            <SCThemeProvider theme={mockTheme}>
                <StatusWrapper status={IN_STOCK} />
            </SCThemeProvider>,
        )
        expect(asFragment()).toMatchSnapshot()
    })
})
