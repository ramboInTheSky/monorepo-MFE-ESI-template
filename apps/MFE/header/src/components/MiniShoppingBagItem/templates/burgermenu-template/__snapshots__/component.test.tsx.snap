// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` components: GridW100PR16  should match the snapshot  1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-template-columns: repeat(2,minmax(0,1fr));
  width: 100%;
  margin-right: 16px;
  font-size: 12px;
}

.c0 .mt-5px {
  margin-top: 5px;
}

.c0 .block {
  display: block;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`StatusWrapper  should match the snapshot with In stock status 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: bold;
  font-size: 12px;
  text-align: right;
  display: block;
  margin-top: 18px;
  color: black;
}

<span
    class="c0"
  />
</DocumentFragment>
`;
